src/linker/Linker.Steps/UnreachableBlocksOptimizer.cs

```diff
@@ -287,186 +216,323 @@ void ProcessStack ()
+				case Code.Ldc_I4_8:
+				case Code.Ldc_I4:
+				case Code.Ldc_I4_S:
+				case Code.Ldc_I4_M1:
+				case Code.Ldc_I8:
+				case Code.Ldc_R4:
+				case Code.Ldc_R8:
+				case Code.Ldnull:
+				case Code.Ldstr:
+					return true;
+				}
+
+				return false;
+			}
+		}
+
+		static bool GetConstantValue (Instruction instruction, out object? value)
+		{
+			switch (instruction.OpCode.Code) {
+			case Code.Ldc_I4_0:
+				value = 0;
+				return true;
+			case Code.Ldc_I4_1:
+				value = 1;
+				return true;
+			case Code.Ldc_I4_2:
+				value = 2;
+				return true;
+			case Code.Ldc_I4_3:
+				value = 3;
+				return true;
+			case Code.Ldc_I4_4:
+				value = 4;
+				return true;
+			case Code.Ldc_I4_5:
+				value = 5;
+				return true;
+			case Code.Ldc_I4_6:
+				value = 6;
+				return true;
+			case Code.Ldc_I4_7:
+				value = 7;
+				return true;
+			case Code.Ldc_I4_8:
+				value = 8;
+				return true;
+			case Code.Ldc_I4_M1:
+				value = -1;
+				return true;
+			case Code.Ldc_I4:
+				value = (int) instruction.Operand;
+				return true;
+			case Code.Ldc_I4_S:
+				value = (int) (sbyte) instruction.Operand;
+				return true;
+			case Code.Ldc_I8:
+				value = (long) instruction.Operand;
+				return true;
+			case Code.Ldstr:
+				value = (string) instruction.Operand;
+				return true;
+			case Code.Ldnull:
+				value = null;
+				return true;
+			default:
+				value = null;
+				return false;
+			}
+		}
+
+		static MethodDefinition? FindSizeMethod (TypeDefinition? type)
+		{
+			if (type == null)
+				return null;
+
+			return type.Methods.First (l => !l.HasParameters && l.IsStatic && l.Name == "get_Size");
+		}
+
+		readonly struct CallInliner
+		{
+			readonly MethodBody body;
+			readonly UnreachableBlocksOptimizer optimizer;
+
+			public CallInliner (MethodBody body, UnreachableBlocksOptimizer optimizer)
+			{
+				this.body = body;
+				this.optimizer = optimizer;
+			}
+
+			public bool RewriteBody ()
+			{
+				bool changed = false;
+				LinkerILProcessor processor = body.GetLinkerILProcessor ();
+				Collection<Instruction> instrs = body.Instructions;
+
+				for (int i = 0; i < body.Instructions.Count; ++i) {
+					Instruction instr = instrs[i];
+					switch (instr.OpCode.Code) {
+
+					case Code.Call:
+					case Code.Callvirt:
+						MethodDefinition? md = optimizer._context.TryResolve ((MethodReference) instr.Operand);
+						if (md == null)
+							continue;
+
+						if (md.IsVirtual)
+							continue;
+
+						if (md.CallingConvention == MethodCallingConvention.VarArg)
 							break;
-						reducer.Rewrite (i, targetResult);
-						changed = true;
-					}
-					break;
 
-				case Code.Sizeof:
-					//
-					// sizeof (IntPtr) and sizeof (UIntPtr) are just aliases for IntPtr.Size and UIntPtr.Size
-					// which are simple static properties commonly overwritten. Instead of forcing C# code style
-					// we handle both via static Size property
-					//
-					MethodDefinition? sizeOfImpl = null;
+						if (md.NoInlining)
```

Comment:
> @MichalStrehovsky might remember more details...

ConstantProp should be blocked for NoInlining methods because they specifically said they don't want to be inlined and it was breaking logic within library build (a constant method that we had a substitution for was getting inlined during library build, breaking the ability to substitute it at app build time).

Isn't constant prop blocked for NoInlining methods because of the check for `if (method.IsIntrinsic () || method.NoInlining)` above anyway? (Would we be able to get a MethodResult for it if we let it past this check here?)

Reply:
You're right - I overlooked it in the existing code - please ignore this comment.